sqlmesh/core/snapshot/definition.py

```diff
@@ -548,39 +559,43 @@ def add_interval(self, start: TimeLike, end: TimeLike, is_dev: bool = False) ->
             self.intervals = merged_intervals
 
     def remove_interval(
-        self, start: TimeLike, end: TimeLike, execution_time: t.Optional[TimeLike] = None
+        self, interval: Interval, execution_time: t.Optional[TimeLike] = None
     ) -> None:
         """Remove an interval from the snapshot.
 
         Args:
-            start: Start interval to remove.
-            end: End interval to remove.
+            interval: The interval to remove.
             execution_time: The date/time time reference to use for execution time. Defaults to now.
         """
-        interval = self.inclusive_exclusive(start, end, execution_time, for_removal=True)
         self.intervals = remove_interval(self.intervals, *interval)
         self.dev_intervals = remove_interval(self.dev_intervals, *interval)
 
-    def inclusive_exclusive(
+    def get_removal_interval(
         self,
         start: TimeLike,
```

Comment:
why is get_removal_interval start, end and remove_interval no longer start and end?

Reply:
I don't really understand. `get_removal_interval` is getting you an interval object (`t.Tuple[int, int]`) while remove_interval is removing an interval object (`t.Tuple[int, int]`). In order to first make an interval you have a start/end and that needs to be translated into an inclusive/exclusive epoch.